#!/usr/bin/env python
PACKAGE = "orb_slam2_ros"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("localize_only",                bool_t,  0, "Disable mapping, localize only",                                                           False)
gen.add("reset_map",                    bool_t,  0, "Erase the map; toggles back to false automatically",                                       False)
gen.add("min_num_kf_in_map",            int_t,   0, "Minimum number of key frames in the map for initialization",                               5, 1, 50)
gen.add("min_observations_for_ros_map", int_t,   0, "Min num of observations per point for the point cloud; Doesn't change the SLAM itself",    4, 2, 30)

exit(gen.generate(PACKAGE, "orb_slam2_ros", "dynamic_reconfigure"))
